devices/HumanLogger/HumanLogger.cpp

```diff
@@ -0,0 +1,463 @@
+            return false;
+        }
+
+        yInfo() << logPrefix << deviceName << "attach() successful";
+    }
+
+    return true;
+}
+
+bool HumanLogger::impl::configureBufferManager()
+{
+    bool ok{true};
+
+    if(settings.logHumanState) {
+        if (loggerType == LoggerType::MATLAB)
+        {
+            jointNamesStateInterface =  iHumanState->getJointNames();
+            ok = ok && bufferManager.addChannel({"human_state::base_position", {3, 1}});
+            ok = ok && bufferManager.addChannel({"human_state::base_orientation", {4, 1}});
+            ok = ok && bufferManager.addChannel({"human_state::base_velocity", {6, 1}});
+            ok = ok && bufferManager.addChannel({"human_state::joint_positions", {jointNamesStateInterface.size(), 1}});
+            ok = ok && bufferManager.addChannel({"human_state::joint_velocities", {jointNamesStateInterface.size(), 1}});
+        }
+    }
+    if (settings.logHumanDynamics)
+    {
+        if (loggerType == LoggerType::MATLAB)
+        {
+            jointNamesDynamicsInterface =  iHumanDynamics->getJointNames();
+            ok = ok && bufferManager.addChannel({"human_dynamics::joint_torques", {jointNamesDynamicsInterface.size(), 1}});
+        }
+    }
+    
+
+    ok = ok && bufferManager.configure(bufferConfig);
+    if (ok) {
+        yDebug() << logPrefix << " buffer manager configured successfully.";
+    }
+
+    return ok;
+}
+
+void HumanLogger::threadRelease() {}
+
+bool HumanLogger::detach()
+{
+    std::lock_guard<std::mutex> guard(pImpl->loggerMutex);
+    while (isRunning()) {
+        stop();
+    }
+
+    pImpl->iHumanState = nullptr;
+    pImpl->iHumanDynamics = nullptr;
+
+    return true;
+}
+
+// ==========================
+// IMultipleWrapper interface
+// ==========================
+
+bool HumanLogger::attachAll(const yarp::dev::PolyDriverList& driverList)
+{
+    bool attachStatus = true;
+
+    if (driverList.size() > 2) {
+        yError() << logPrefix << "This wrapper accepts maximum three attached PolyDriver.";
+        return false;
+    }
+
+    for (size_t i = 0; i < driverList.size(); i++) {
+        const yarp::dev::PolyDriverDescriptor* driver = driverList[i];
+
+        if (!driver) {
+            yError() << logPrefix << "Passed PolyDriverDescriptor is nullptr";
+            return false;
+        }
+
+        attachStatus = attachStatus && attach(driver->poly);
+    }
+
+    if (!pImpl->configureBufferManager()) {
+        yError() << logPrefix << "Failed to configure buffer manager for the logger.";
+    }
+
+    if (attachStatus) {
+        yDebug() << logPrefix << "attach() successful";
+    }
+
+    // ====
+    // MISC
+    // ====
+
+    // Start the PeriodicThread loop
+    if (attachStatus && !start()) {
+        yError() << logPrefix << "Failed to start the loop.";
+        return false;
+    }
+
+    return attachStatus;
+}
+
+bool HumanLogger::detachAll()
+{
+    return detach();
+}
```

Comment:
It's always better to have blank lines at EOF (see also https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)